### ISSUE TITLE
media-type conversion broken?

### ISSUE THREAD

**Issue by woutor** [willnorris/imageproxy#173]
Hi,

I recently updated image proxy and found out that converting the media type (f.e. `/620x0,jpeg/images/OFLmfUmkssPmyH67.png`) returns the following error message: `requested URL is not allowed`. The log says `content-type not allowed: ""`.

I was able to work around the problem by adding ` -contentTypes ""` as an argument to imageproxy. However, I was wondering whether this is expected behaviour?

**Comment by willnorris** [comment]
Well, it is expected behavior in that content-type filtering was added in the last release, and the default value for the `-contentType` flag is "image/*" (more details [here](https://github.com/willnorris/imageproxy#allowed-content-type-list)).

Though I'm a little curious why the content-type of the image is empty.  Are you using the latest release (0.8.0) or are you building from the master branch.  If the latter, do you know what commit?  I added content-type detection a few days ago in cf54b2c.  Try updating to the latest master, and see if it's still a problem?

Alternately, could you provide the full URL of the proxied image? I'm curious if the upstream server is misconfigured and not sending a Content-Type header.

**Comment by woutor** [comment]
Hi @willnorris,

I am running the 0.8.0 release from docker hub. The actual image is [here](https://s3.eu-central-1.amazonaws.com/groovetrackers.com/images/OFLmnUmkssb7KthPmyH67C7d.png) (I shortened the URL in the first post). I checked the response headers and S3 does provide a Content-Type header. 

I am also using `https://s3.eu-central-1.amazonaws.com/groovetrackers.com/` as a baseURL. I noticed you were working on the Content-Type detection, but these changes have not been released to docker hub. 

Just to be sure, I also checked out master:
```
go build willnorris.com/go/imageproxy/cmd/imageproxy
cp imageproxy /usr/local/bin
export BASE_URL='https://s3.eu-central-1.amazonaws.com/groovetrackers.com/'
export PORT=8080 
imageproxy -addr 0.0.0.0:$PORT -baseURL $BASE_URL
```
Then visited `http://localhost:8080/620x0,sc,jpeg/images/OFLmnUmkssb7KthPmyH67C7d.png` with the same error. If I remove the jpeg conversion argument from the URL it works as expected.

**Comment by yurikburko** [comment]
Hello,
Have the same issue using docker image with latest tag (currently it is v0.8.0) from docker hub.
I am sure that server is sending a Content-Type header for requested image ("image/*" content type), but imageproxy responses with 403 status code and logs  _content-type not allowed: ""_ message.

**Comment by mezoistvan** [comment]
We also have the same issue where the picture's response headers are lowercase... maybe this `Header.Get` is case sensitive?https://github.com/willnorris/imageproxy/blob/d4246a08fdec341ddf01b09e74e56ee03f9929d0/imageproxy.go#L191

**Comment by willnorris** [comment]
@yurikburko if you're still having this problem, would you mind sharing the remote URL that imageproxy is throwing a 403 on?  Also, can you provide the flags you're passing to imageproxy on startup?

**Comment by willnorris** [comment]
closing as fixed by cf54b2c.  If anyone continues to see this problem, please feel free to re-open.

**Comment by cgwyllie** [comment]
@willnorris we've encountered a very similar issue with the 0.9.0 release, although the issue is intermittent, only seeming to affect certain dimensions of resizing, and with certain images when format conversion is enabled.

When conversion is not requested, the images are resized as expected.

Here are the details to reproduce for us:

- Check out the `v0.9.0` tag
- Build the imageproxy server command
- Run with `./imageproxy -cache memory:200:24h -baseURL https://media-cdn.app.delivery/ -allowHosts media-cdn.app.delivery -scaleUp true`
- Request `http://localhost:8080/16,png/https:/media-cdn.app.delivery/kloud.png` from a browser

Building from master does not exhibit the issue.

Whilst we briefly tried to isolate which commit fixes it, we didn't spend much time on it and couldn't pin it to a specific one.

For us, we have built & deployed from master, but wanted to pass on the details in case they're of any use. Is there another release planned?

Thanks,
Chris
